src/dmd/mars.d

```diff
@@ -242,18 +242,25 @@ private int tryMain(size_t argc, const(char)** argv)
      */
     sections.push("Environment");
     parseConfFile(&environment, global.inifilename, inifilepath, inifile.len, inifile.buffer, &sections);
-    Strings dflags;
-    getenv_setargv(readFromEnv(&environment, "DFLAGS"), &dflags);
-    environment.reset(7); // erase cached environment updates
+
     const(char)* arch = global.params.is64bit ? "64" : "32"; // use default
     arch = parse_arch_arg(&arguments, arch);
-    arch = parse_arch_arg(&dflags, arch);
+
+    // parse architecture from DFLAGS read from [Environment] section
+    {
+        Strings dflags;
+        getenv_setargv(readFromEnv(&environment, "DFLAGS"), &dflags);
+        environment.reset(7); // erase cached environment updates
+        arch = parse_arch_arg(&dflags, arch);
+    }
+
```

Comment:
`dflags` is never used after this point, and will actually contain incomplete information after the `[Environment{arch}]` section is parsed, so it is placed in it's own scope.